src/freenas/usr/local/lib/middlewared_truenas/plugins/failover.py

```diff
@@ -1864,7 +1864,7 @@ async def _event_system_ready(middleware, event_type, args):
         return
 
     if await middleware.call('keyvalue.get', 'HA_UPGRADE', False):
-        middleware.send_event('failover.upgrade_pending', 'ADDED', id='BACKUP', fields={'pending': True})
+        middleware.send_event('failover.upgrade_pending', 'ADDED', fields={'id': 'BACKUP', 'pending': True})
```

Comment:
Thats not direct, I will need some more explanation. What are the consumers of this event?

Reply:
If you mean 

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/middlewared/main.py", line 1343, in wrap
    await handler(self, event_type, kwargs)
  File "/usr/local/lib/middlewared_truenas/plugins/failover.py", line 1865, in _event_system_ready
    middleware.send_event('failover.upgrade_pending', 'ADDED', {
```

this has already been fixed in 963c3ae442b996a9378c774271a6c3d81503bad2